pytradfri/resource.py

```diff
@@ -65,11 +65,13 @@ class ApiResourceResponse(BaseModel):
     ota_update_state: int | None = Field(alias=ATTR_OTA_UPDATE_STATE)
     reachable: int | None = Field(alias=ATTR_REACHABLE_STATE)
 
-    air_purifier: list[AirPurifierResponse] | None = Field(alias=ROOT_AIR_PURIFIER)
-    blind: list[BlindResponse] | None = Field(alias=ATTR_START_BLINDS)
-    light: list[LightResponse] | None = Field(alias=ATTR_LIGHT_CONTROL)
-    socket: list[SocketResponse] | None = Field(alias=ATTR_SWITCH_PLUG)
-    signal_repater: list[SignalRepeaterResponse] | None = Field(
+    air_purifier_control: list[AirPurifierResponse] | None = Field(
```

Comment:
These attributes should go under a `DeviceResponse` model.

Reply:
Here's how I would write the `DeviceResponse` model:

```py
class DeviceResponse(ApiResourceResponse):
    """Represent a device response."""

    # Type with Any for now to allow smaller typing work chunks

    air_purifier_control: Optional[List[AirPurifierResponse]] = Field(
        alias=ROOT_AIR_PURIFIER
    )
    application_type: int = Field(alias=ATTR_APPLICATION_TYPE)
    blind_control: Optional[List[BlindResponse]] = Field(alias=ATTR_START_BLINDS)
    device_info: DeviceInfoResponse = Field(alias=ATTR_DEVICE_INFO)
    last_seen: Optional[int] = Field(alias=ATTR_LAST_SEEN)
    light_control: Optional[List[Dict[str, Any]]] = Field(alias=ATTR_LIGHT_CONTROL)
    reachable_state: int = Field(alias=ATTR_REACHABLE_STATE)
    signal_repeater_control: Optional[List[Dict[str, Any]]] = Field(
        alias=ROOT_SIGNAL_REPEATER
    )
    socket_control: Optional[List[Dict[str, Any]]] = Field(alias=ATTR_SWITCH_PLUG)
```

Reply:
Thank you!